x-pack/plugins/enterprise_search/server/lib/pipelines/create_pipeline_definitions.test.ts

```diff
@@ -193,7 +173,19 @@ describe('formatMlPipelineBody util function', () => {
       destField,
       mockClient as unknown as ElasticsearchClient
     );
-    expect(actualResult).toEqual(expectedResult);
+    expect(actualResult).toEqual(
+      expect.objectContaining({
+        processors: expect.arrayContaining([
+          {
+            inference: expect.objectContaining({
+              field_map: {
+                [sourceField]: modelInputField,
+              },
+            }),
+          },
+        ]),
+      })
+    );
```

Comment:
I think it's unnecessary code duplication if we check the entire result object here - this is already covered in the "should return the pipeline body" test case. I modified the test to only look for the field that is different under the actual conditions.

Reply:
While it is quite easy to do unwanted side effects, we usually do a full object checks especially for the Redux/Kea states to make sure we _only_ change the part where we want to change. I am curious if that would be useful here too.

Reply:
Hm, good point Efe. I'll update the code to work with full objects and `assign()` the modified part that the test is focusing on.

Reply:
You can use destructuring as well. 

i.e.
```typescript
{
  ...EXPECTED_DEFAULTS,
  only: {
    the_changed: "parts"
  }
}
```

Reply:
I ended up using lodash `merge()` since destructuring didn't work for changing a single array element's specific property while leaving everything else untouched. (Or maybe I just don't know how to do it 🙂)